avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -752,6 +752,9 @@ public SchemaAndValue toConnectData(org.apache.avro.Schema avroSchema, Object va
   }
 
   private Object toConnectData(Schema schema, Object value) {
+    if (value == null) {
+      return null;
```

Comment:
Ok, added that as well including a test case.
Please note that this is however an "api" change: previously this invalid null-value fields would pass through, now they will throw the DataException.